fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java

```diff
@@ -104,4 +122,28 @@ public void buildAuthSwitchRequest(MysqlSerializer serializer) {
         serializer.writeBytes(authPluginData);
         serializer.writeInt1(0);
     }
+
+    // If user use kerberos for authentication, fe need to resend the handshake request.
+    public void buildKrb5AuthRequest(MysqlSerializer serializer, String remoteIp, String user) throws Exception {
+        String fullUserName = SystemInfoService.DEFAULT_CLUSTER + ":" + user;
+        Password password = Catalog.getCurrentCatalog().getAuth().getUserPrivTable()
+                .getPasswordByApproximate(fullUserName, remoteIp);
+        if (password == null) {
+            String msg = String.format("Can not find password with [user: %s, remoteIp: %s].", user, remoteIp);
+            LOG.error(msg);
+            throw new Exception(msg);
+        }
+
+        String userRealm = password.getUserForAuthPlugin();
+        File jarFile = new File(KRB5_AUTH_JAR_PATH);
+        ClassLoader loader = URLClassLoader.newInstance(
+                new URL[] { jarFile.toURL() },
+                getClass().getClassLoader()
+        );
```

Comment:
Better to load the class only once. One option is to create an interface in fe, and the implementation of the interface is in the commercial version of the jar package.

Reply:
done